binaries/oli/src/config/mod.rs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use opendal::{services, Operator, Scheme};
+use serde::Deserialize;
+use toml;
+
+type StringMap<T> = HashMap<String, T>;
+
+macro_rules! update_options {
```

Comment:
maybe we can remove this macro rule?

Reply:
This purpose of this macro is to save me some typings, otherwise I need to execute these statements in every case listed in https://github.com/apache/incubator-opendal/blob/8439153fbedcf1b303cac85befed7dd4b2e9b7de/binaries/oli/src/config/mod.rs#L98-L105
which I think is repetitive

Reply:
don't worry, from_map handles this.

Reply:
well.. I have to admit it makes sense 😂 will do